juriscraper/oral_args/united_states/state/miss.py

```diff
@@ -0,0 +1,135 @@
+"""Scraper for [Full name of court]
+CourtID: [unique abbreviation to be used by software/filesystem]
+Court Short Name: [standard abbreviation used in citations]
+Author:
+Reviewer:
+History:
+  YYYY-MM-DD: Created by XXX
+"""
+import os
+import calendar
+
+from lxml import html
+from juriscraper.OralArgumentSite import OralArgumentSite
+from juriscraper.lib.string_utils import titlecase
+from juriscraper.lib.string_utils import convert_date_string
+
+DAYS = [d.lower() for d in calendar.day_name]
+
+def get_first_word(text):
+    return text.split()[0]
+
+
+def make_year_iterable(year):
+    "returns a (year, sitting) tuple"
+    iterable = list()
+    for sitting in range(1, 7):
+        iterable.append((year, sitting))
+    return iterable
+
+
+def make_back_scrape_iterable():
+    iterable = [(2014, 5), (2014, 6)]
+    for year in range(2015, 2019):
+        iterable += make_year_iterable(year)
+    return iterable
+
+
+def parse_sibling_text_nodes(iframe):
+    textlist = iframe.xpath("./following-sibling::text()")
+    textlist = [el.strip() for el in textlist if el.strip()]
+    text_nodes = list()
+    for tnode in textlist:
+        if tnode != '\\r\\n':
+            text_nodes.append(tnode)
+    if len(text_nodes) > 1:
+        raise RuntimeError("Parsing error on {}".format(iframe))
+    return text_nodes
+
+
+def parse_emboldened_nodes(iframe):
+    # FIXME test './following-sibling::b|strong'
+    # or variants
+    boldlist = iframe.xpath('./following-sibling::b')
+    if not len(boldlist):
+        boldlist = iframe.xpath('./following-sibling::strong')
+    if not len(boldlist):
+        msg = "No emboldened node in {}".format(iframe.getparent())
+        raise RuntimeError(msg)
+    date_text = None
+    while len(boldlist):
+        el = boldlist.pop(0)
+        text = el.text_content()
+        # Expecting date string to start with capitalized day name
+        fword = get_first_word(text)
+        if fword.endswith(','):
+            fword = fword.split(',')[0]
+        if fword.lower() not in DAYS:
+            continue
+        else:
+            date_text = text
+    return date_text
+
+
+class Site(OralArgumentSite):
+    def __init__(self, *args, **kwargs):
+        super(Site, self).__init__(*args, **kwargs)
+        self.court_id = self.__module__
+        self._url_template = "https://courts.ms.gov/appellatecourts/{court}/archive/{year}/{scourt}{sitting}{year}.php"  # noqa: E501
+        self.method = 'GET'
+        self.uses_selenium = False
+        # Complete this variable if you create a backscraper.
+        self.back_scrape_iterable = make_back_scrape_iterable()
+        self.url = "https://courts.ms.gov/appellatecourts/sc/scoa.php"
+
+    def _make_url(self, year, sitting):
+        data = dict(year=year, sitting=sitting)
+        court_name = dict(court="sc", scourt="scs")
+        data.update(court_name)
+        return self._url_template.format(**data)
+
+    def _get_download_urls(self):
```

Comment:
The lines you marked are for setting the url for the sittings page that holds the videos.  Also, to answer third bullet above, iframes are selected only because it verifies that a video link is available.  The iframes are not followed, but merely selected to skip parsing the empty pages (sittings when the court doesn't hold arguments).  The actual html being parsed are siblings to the iframe in each ```td``` that contains an iframe.

Reply:
> The lines you marked are for setting the url for the sittings page that holds the videos.

Sure, but you can still use the `@href` part here rather than using the el.get('href') that you have later. Should simplify things.

Reply:
done